src/routes/(app)/product/[id]/+page.server.ts

```diff
@@ -166,18 +167,37 @@ async function addToCart({ params, request, locals }: RequestEvent) {
 			chosenVariations: z.record(z.string(), z.string()).optional()
 		})
 		.parse(json);
+	const variationPrice =
+		product.hasVariations && chosenVariations
+			? sumCurrency(
+					runtimeConfig.mainCurrency,
+					Object.entries(chosenVariations).map((variation) => ({
+						amount:
+							product.variations?.find(
+								(vari) => variation[0] === vari.name && variation[1] === vari.value
+							)?.price ?? 0,
+						currency: runtimeConfig.mainCurrency
+					}))
+			  )
+			: 0;
 
 	const customPrice =
 		customPriceAmount && customPriceCurrency
 			? {
-					amount: parsePriceAmount(customPriceAmount, customPriceCurrency),
+					amount: Number(customPriceAmount),
```

Comment:
Revert? We do want to call `parsePriceAmount` when the user adds a custom price for a product that allows it.